### DISCUSSION TITLE
Alert based PCAP

### DISCUSSION THREAD

**Discussion by xfaith** [Security-Onion-Solutions/securityonion#7938] [Unsupported Versions] [answered]
Is there a way to say for SecurityOnion to only save  PCAP based on Suricata Alerts?  I have read that you can configure BPF to stop Stenographer from saving PCAP to disk-based on certain criteria but wanted to see if there is a way, or maybe future plans for something alert-based.

**Comment by TotieBash** [comment]
You can use Suricata to do PCAP capture on triggered alerts and stop stenographer all together.  Do you need Zeek metadata?  Because if all you need is Suricata features then you probably dont need a full SO install.

**Comment by dougburks** [comment, marked as answer]
The problem with alert-based PCAP is that in the best case scenario the NIDS alerts only represent a very small percentage of the traffic in the incident and so your PCAP will be limited. In the worst case scenario, the attacker evades your NIDS alerts and so you have no PCAP whatsoever.

It has been our experience that it is better to keep PCAP independent of NIDS alerts. If your disk space is constrained, there are other ways of limiting what PCAP is stored while still being effective in incidents. Start with excluding port 443 and other connections where PCAP is less useful. Another option might be only recording PCAP for high value hosts or highly targeted hosts.

**Comment by xfaith** [comment]
Thanks everyone for the responses, agree with the excluding some traffic like suggested (currently working out that part), just want to make sure I have all the information for upcoming discussions on possible ways forward.
